\section{Introduction}

Phylogenetics is rapidly progressing as the statistical foundation of
comparative biology, providing a framework for accounting for the shared
ancestry inherent in biological data.
Soon after phylogenetics became feasible as a likelihood-based statistical
endeavor \citep{Felsenstein1981}, models started becoming richer to better
capture processes of biological diversification and character change.
This trend in model complexity made Bayesian approaches appealing, because they
can render inference under rich models practical by leveraging prior
information and numerical techniques (e.g., data augmentation).

From the earliest days of Bayesian phylogenetics \citep{Rannala1996,Mau1997},
the numerical tool of choice for approximating the posterior distribution was
Markov chain Monte Carlo (MCMC).
By only considering the ratios of posterior densities, MCMC obviates the need
for calculating the proportionality constant in the denominator of Bayes' rule.
This proportionality constant is better known as the marginal likelihood---the
probability of the data under the model, averaged, with respect to the prior,
over the whole parameter space.
This marginalized measure of model fit cannot be solved analytically due to
the large number of parameters in phylogenetic models that need to be summed or
integrated.

Marginal likelihoods are central to model comparison in a Bayesian framework.
Thus, we cannot avoid calculating them if we want to be able to compare the fit
of phylogenetic models.
As the diversity and richness of phylogenetic models has increased, there has
been a renewed appreciation of the importance of such Bayesian model
comparison.
As a result, there has been a lot of work over the last decade to develop
methods for estimating marginal likelihoods of phylogenetic models.

The goals of this review are to
(1) provide a summary about what marginal likelihoods are and why they are
useful,
(2) review the various methods available for approximating marginal likelihoods
of phylogenetic models,
(3) review applications
(4) \ldots

\section{What is a marginal likelihood?}

\begin{linenomath}
A marginal likelihood likelihood is the average fit of a model to a dataset.
More specifically, it is an average over the entire parameter space of the
likelihood weighted by the prior.
For a phylogenetic model \model with parameters that include the discrete
topology (\topology) and continous branch lengths and parameters that govern
the evolution of the characters along the tree (together represented by
\evoparameters), the marginal likelihood can be represented as
\begin{equation}
    p(\data \given \model) =
    \sum\limits_{\topology}
    \int_{\evoparameters}
    p( \data \given \topology, \evoparameters, \model)
    p(\topology, \evoparameters \given \model)
    \diff{\evoparameters},
    \label{eq:marginalLikelihood}
\end{equation}
where \data are the data.
Each parameter in \evoparameters adds a dimension to the model, over which the
likelihood must be averaged.
\end{linenomath}

\begin{linenomath}
The marginal likelihood is also the proportionality constant in the denominator
of Bayes' rule that ensures the posterior is a proper probability density that
integrates to one:
\begin{equation}
    p(\topology, \evoparameters \given \data, \model) = \frac{
        p(\data \given \topology, \evoparameters, \model)
        p(\topology, \evoparameters \given \model)
    }{
        p(\data \given \model)
    }.
    \label{eq:bayesRule}
\end{equation}
\end{linenomath}

\section{Why are marginal likelihoods useful}

Marginal likelihoods are the currency of model comparison in a Bayesian
framework. The frequentist approach to model choice is based on comparing
the maximum probability of the data under two models either using
a likelihood ratio test or some information-theoretic criterion.
Because adding a parameter (dimension) to a model will always ensure
a maximum likelihood at least as large as without the parameter, some
penalty must be imposed when parameters are added.

In a Bayesian perspective, we are interested in comparing the average fit of a
model, rather than the maximum.
This imposes a ``natural'' penalty for parameters, because each additional
parameter introduces a dimension that must be averaged over.
If that dimension introduces a lot of parameter space with small likelihood,
and little space that improves the likelihood, it will decrease the marginal
likelihood.
Thus, unlike the maximum likelihood, adding a parameter of a model can
decrease the marginal likelihood.

The ratio of two marginal likelihoods gives us the factor by which the
average fit of the model in the numerator is better or worse than the
model in the denominator.
This is called the Bayes' factor, and is also equal to the factor by
which the data updated the prior odds of the model.
This gives us an intuitive measure how much the data ``favor'' one model over
another.

\section{How to approximate a marginal likelihood}

For all but the simplest of models, the integration in
Equation~\ref{eq:marginalLikelihood}
is analytically intractable.
This is particularly true for phylogenetic models
For phylogenetic models, the integration in
Equation~\ref{eq:marginalLikelihood}, which entails averaging the likelihood
over all topologies, the branch lengths of every branch, and all other model
parameters, is analytically intractable.
Perhaps the simplest numerical approximation of the marginal likelihood is to
draw samples of a model's parameters from their respective prior distributions.
This turns the intractable integral into a sum of the samples' likelihoods.
Because the prior weight of each sample is one in this case, the marginal
likelihood can be approximated by simply calculating the average likelihood of
the prior samples.
Similarly, if we have a sample of the parameters from the posterior
distribution---like one obtained from a ``standard'' Bayesian phylogenetic
analysis via MCMC---we can again use summation to approximate the integral.
The weight of each sample is the ratio of the prior density to the posterior
density.
As a result, the sum simplifies to the harmonic mean (HM) of the likelihoods
from the posterior sample \citep{Newton1994}.
Both of these techniques are importance-sampling approximations, and suffer
from the fact that the prior and posterior are often \emph{very} divergent,
with the latter usually \emph{much} more peaked than the former.
A finite sample from the prior will often yield an underestimate of the
marginal likelihood, because the region of parameter space with high likelihood
is likely to be missed.
Whereas a finite sample from the posterior will almost always lead to an
overestimate, because it will contain very few samples outside of the region of
high likelihood, where the prior has a strong downward ``pull'' on the average
likelihood.

Recent methods developed to estimate marginal likelihoods generally fall into
two categories of how to deal with the sharp contrast between the prior
and posterior that cripples the simple importance-sampling approaches
mentioned above.
One general strategy is to turn the giant leap between the unnormalized
posterior and prior into many small steps along a series of power posterior
distributions.
The second strategy is to turn the giant leap between the posterior and prior
into a smaller leap between the posterior and a reference distribution that is
as similar as possible to the posterior.
These approaches are not mutually exclusive (e.g., see Fan et al.\
\citeyear{Fan2011}), but they serve as a useful way to categorize marginal
likelihood approximation methods.
In practical terms, the first strategy is computationally expensive, because
samples need to be collected from each of the power posterior distributions,
which is not part of a ``standard'' Bayesian phylogenetic analysis.
The second strategy is very inexpensive because it attempts to approximate the
marginal likelihood using only the posterior samples collected from a
``standard'' analysis.

\subsection{Approaches that use only posterior samples}

\begin{itemize}
    \item Generalized harmonic mean (GHM) \citep{Gelfand1994}
    \item Inflated density ratio \citep{Arima2012}
    \item Partition weighted kernel (PWK) \cite{Wang2017}; Looks promising, but
        not used for phylo yet.
        \begin{itemize}
            \item Entails partitioning parameter space into regions within which the
                posterior density is relatively homogeneous.
            \item Given the complex structure of phylogenetic models
                (continuous and discrete elements; cite hyper-dimensional
                orange), it's not clear how this would be done.
        \end{itemize}
\end{itemize}

\subsection{Approaches that use power posterior samples}

\begin{itemize}
    \item Path-sampling \citep{Lartillot2006}
    \item Stepping-stone \citep{Xie2011}
    \item Generalized stepping stone \citep{Fan2011}
\end{itemize}

\subsection{Importance sampling-ish}

\begin{itemize}
    \item Sequential stochastic approximation Monte Carlo (SSAMC)
        \citep{Liang2007,Cheon2008}
    \item Sequential Monte Carlo (SMC) \citep{Jordan2012}
    \item Direct sampling (model averaging)
    \item Approximate-likelihood Bayesian approaches. Wegmann's ABCToolbox
        approximates the posterior distribution with a GLM, and then takes the
        marginal of that distribution. Knowles' lab is using this to compare
        different phylogeographic models. It would be interesting to know how
        accurate these marginal likelihoods are (my guess is that they are are
        not so good). As far as I'm aware there have been no analyses to assess
        this.
        \thought{Could perform Christian Robert-esque assessment of the
            accuracy and precision of GLM approximations of the marginal
            probability of the data.}
\end{itemize}

\subsection{Approximate-likelihood approaches}

\begin{itemize}
    \item Approximate-likelihood Bayesian approaches. Wegmann's ABCToolbox
        approximates the posterior distribution with a GLM, and then takes the
        marginal of that distribution. Knowles' lab is using this to compare
        different phylogeographic models. It would be interesting to know how
        accurate these marginal likelihoods are (my guess is that they are are
        not so good). As far as I'm aware there have been no analyses to assess
        this.
\end{itemize}

\section{Uses of marginal likelihoods}

\subsection{Species delimitation}

Bayes factor delimitation in *BEAST and SNAPP.

\subsection{Comparing partitioning schemes}

Brandley and others used importance sampling (harmonic mean estimator) to
compare the fit of different ways to partition sequence alignments into rate
categories.

\subsection{Comparing nucleotide substitution models}

\subsection{Comparing ``relaxed clock'' models}

\subsection{Comparing tree models (priors)}

\subsection{Comparing phylogeographic models}

Knowles' lab is using Weggman's GLM approximation to compare the fit of
phylogeographic models built atop different ecological nich models.

Search for other uses of the GLM marginal.
